django/conf/global_settings.py

```diff
@@ -492,6 +492,7 @@
 PASSWORD_HASHERS = [
     'django.contrib.auth.hashers.PBKDF2PasswordHasher',
     'django.contrib.auth.hashers.PBKDF2SHA1PasswordHasher',
+    'django.contrib.auth.hashers.Argon2PasswordHasher',
```

Comment:
If we would all pick sufficiently strong password, than `SHA256` would be a perfectly fine password hash.  The current default setting for PBKDF2 on Django, uses 60,000 iterations of `SHA256`.  Thus if an attacker would have been able to crack your stolen SHA256 hashes in a day, he would have to wait 164 years with PBKDF2.

A capable attacker will use a GPU, a FPGA or maybe even an ASIC to brute force PBKDF2.  Per dollar an ASIC is in the order of 50,000 times more efficient for SHA256, than a CPU.  (Check MHash/J between [ASIC bitcoin miners](https://en.bitcoin.it/wiki/Mining_hardware_comparison) and [CPU bitcoin miners](https://en.bitcoin.it/wiki/Non-specialized_hardware_comparison).)  argon2 (and the other new password hashes) try to close this gap: they are designed such that they shouldn't be a lot faster to compute on an ASIC than on a general purpose CPU.  Thus, if there are no weaknesses in argon2, and we take parameters such that it runs as quick as PBKDF2, then a password which could have been brute forced in a day if PBKDF2 was used, would take in the order of centuries with argon2.

I do agree: PBKDF2 is a fine password hash.  However, I wanted to emphasise that argon2 promises to be a real improvement instead of a marginal one.